src/server/routes/signIn.ts

```diff
@@ -405,6 +406,48 @@ router.post(
   }),
 );
 
+/**
+ * If an Okta session exists, this route will re-authenticate the Okta session
+ * and also refresh the IDAPI session concurrently, synchronising the expiry
+ * times of the Okta and IDAPI sessions, before returning the client to the URL
+ * they came from.
+ */
+router.get(
+  '/signin/refresh',
+  handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
+    const { useOkta, returnUrl } = res.locals.queryParams;
+    const oktaSessionCookieId: string | undefined = req.cookies.sid;
+    const identitySessionCookie = req.cookies.SC_GU_U;
+    const identityLastAccessCookie = req.cookies.SC_GU_LA;
+
+    const { defaultReturnUri } = getConfiguration();
+    const redirectUrl = returnUrl || defaultReturnUri;
+
+    // Check if the user has an existing Okta session.
+    if (okta.enabled && useOkta && oktaSessionCookieId) {
+      try {
+        // If the user session is valid, we re-authenticate them, supplying
+        // the SID cookie value to Okta.
+        await getSession(oktaSessionCookieId);
+        return performAuthorizationCodeFlow(req, res);
+      } catch {
+        //if the cookie exists, but the session is invalid, we remove the cookie
+        //and return them to the URL they came from
+        clearOktaCookies(res);
+        return res.redirect(redirectUrl);
+      }
+    } else if (identitySessionCookie && identityLastAccessCookie) {
```

Comment:
```suggestion
    } else if (identitySessionCookie) {
```

we shouldn't check for the `SC_GU_LA` cookie in this flow, as that cookie is only ever set for 30 mins, so most users won't have this cookie anyway when logged in